packages/web3-providers/src/resolvers/ProviderResolver.js

```diff
@@ -74,6 +75,10 @@ export default class ProviderResolver {
             return provider;
         }
 
+        if (provider instanceof Web3EthereumProvider) {
```

Comment:
Could you change this check to: ``if(provider.host == 'Web3EthereumProvider')`` beacuse this is 100% mangle save :)

Reply:
Sounds good! fixed:  https://github.com/ethereum/web3.js/pull/2856/commits/9194f4146b93621a412b8ce851eda838da88ca06